static/panes/cfg-view.ts

```diff
@@ -98,125 +85,79 @@ export class Cfg {
-                        });
+                    this.currentFunc = val;
+                    this.showCfgResults({
+                        nodes: selectedFn.nodes,
+                        edges: selectedFn.edges,
+                    });
+                    if (selectedFn.nodes && selectedFn.nodes.length > 0) {
                         this.cfgVisualiser.selectNodes([selectedFn.nodes[0].id]);
-                        this.resize();
-                        this.saveState();
                     }
-                },
-            } as any
-            // The current version of tom-select has a very restrictive type definition
-            // that forces to pass the whole options object. This is a workaround to make it type check
-        );
-
-        this.paneRenaming = new PaneRenaming(this, state);
+                    this.resize();
+                    this.updateState();
+                }
+            },
+        });
 
-        this.initCallbacks();
         this.updateButtons();
+    }
+
+    override getInitialHTML(): string {
+        return $('#cfg').html();
+    }
+
+    override registerOpeningAnalyticsEvent(): void {
         ga.proxy('send', {
             hitType: 'event',
             eventCategory: 'OpenViewPane',
             eventAction: 'Cfg',
         });
     }
 
-    onCompileResult(compilerId: number, compiler: any, result: any) {
-        if (this.compilerId === compilerId) {
+    override onCompileResult(compilerId: number, compiler: any, result: any) {
+        if (this.compilerInfo.compilerId === compilerId) {
             let functionNames: string[] = [];
-            if (this.supportsCfg && !$.isEmptyObject(result.cfg)) {
+            if (compiler.supportsCfg && !$.isEmptyObject(result.cfg)) {
                 this.functions = result.cfg;
                 functionNames = Object.keys(this.functions);
                 if (functionNames.indexOf(this.currentFunc) === -1) {
                     this.currentFunc = functionNames[0];
                 }
+                const selectedFn = this.functions[this.currentFunc];
                 this.showCfgResults({
-                    nodes: this.functions[this.currentFunc].nodes,
-                    edges: this.functions[this.currentFunc].edges,
+                    nodes: selectedFn.nodes,
+                    edges: selectedFn.edges,
                 });
-                this.cfgVisualiser.selectNodes([this.functions[this.currentFunc].nodes[0].id]);
+                if (selectedFn.nodes && selectedFn.nodes.length > 0) {
+                    this.cfgVisualiser.selectNodes([selectedFn.nodes[0].id]);
+                }
             } else {
                 // We don't reset the current function here as we would lose the saved one if this happened at the beginning
                 // (Hint: It *does* happen)
-                if (result.compilationOptions.indexOf('-emit-llvm') === -1) {
+                if (!result.compilationOptions?.includes('-emit-llvm')) {
```

Comment:
Found this by accident, but if the compilation fails because of server issue (In my case, I had an ENOENT on my c++filt), `result.compilationOptions` would be undefined. This handles that unlikely case (Instead of just showing the error on the compiler editor), but I'm guessing there are more instances of this spread around the code :/